packages/spacecat-shared-data-access/src/models/importer/import-job.js

```diff
@@ -0,0 +1,178 @@
+
+  /**
+     * Updates the end time of the ImportJob.
+     * @param {string} endTime - The new end time.
+     * @returns {ImportJob} The updated ImportJob object.
+     */
+  self.updateEndTime = (endTime) => {
+    if (!isIsoDate(endTime)) {
+      throw new Error(`Invalid end time during update: ${endTime}`);
+    }
+
+    self.state.endTime = endTime;
+    self.touch();
+
+    return self;
+  };
+
+  /**
+     * Updates the duration of the ImportJob.
+     * @param {number} duration - The new duration.
+     * @returns {ImportJob} The updated ImportJob object.
+     */
+
+  self.updateDuration = (duration) => {
+    if (typeof duration !== 'number' || duration < 0) {
+      throw new Error(`Invalid duration during update: ${duration}`);
+    }
+
+    self.state.duration = duration;
+    self.touch();
+
+    return self;
+  };
+
+  /**
+     * Updates the status of the ImportJob.
+     * @param {string} status - The new status.
+     * @returns {ImportJob} The updated ImportJob object.
+     */
+
+  self.updateStatus = (status) => {
+    if (!Object.values(IMPORT_JOB_STATUS).includes(status)) {
+      throw new Error(`Invalid Import Job status during update: ${status}`);
+    }
+
+    self.state.status = status;
+    self.touch();
+
+    return self;
+  };
+
+  /**
+     * Updates the success count of the ImportJob.
+     * @param {number} successCount - The new success count.
+     * @returns {ImportJob} The updated ImportJob object.
+     */
+  self.updateSuccessCount = (successCount) => {
+    if (typeof successCount !== 'number' || successCount < 0) {
+      throw new Error(`Invalid success count during update: ${successCount}`);
+    }
+
+    self.state.successCount = successCount;
+    self.touch();
+
+    return self;
+  };
+
+  /**
+     * Updates the failed count of the ImportJob.
+     * @param {number} failedCount - The new failed count.
+     * @returns {ImportJob} The updated ImportJob object.
+     */
+  self.updateFailedCount = (failedCount) => {
+    if (typeof failedCount !== 'number' || failedCount < 0) {
+      throw new Error(`Invalid failed count during update: ${failedCount}`);
+    }
+
+    self.state.failedCount = failedCount;
+    self.touch();
+
+    return self;
+  };
+
+  /**
+     * Updates the import queue id of the ImportJob.
+     * @param {string} importQueueId - The new import queue id.
+     * @returns {ImportJob} The updated ImportJob object.
+     */
+  self.updateImportQueueId = (importQueueId) => {
+    if (typeof importQueueId !== 'string' || importQueueId.length === 0) {
+      throw new Error(`Invalid import queue id during update: ${importQueueId}`);
+    }
+
+    self.state.importQueueId = importQueueId;
+    self.touch();
+
+    return self;
+  };
+  return Object.freeze(self);
+};
+
+/**
+ * Creates a new ImportJob object.
+ * @param {Object} importJobData - The data for the ImportJob object.
+ * @returns {ImportJob} The new ImportJob object.
+ */
+
+export const createImportJob = (data) => {
```

Comment:
Can the validations above be reused so they are not duplicated?

Reply:
I modified the validation for apiKey to be `!isString(apiKey)`.
I'm using spacecat-shared utils for validation.